client/components/user/Settings/UserGeneral.vue

```diff
@@ -0,0 +1,194 @@
+            @file-choose="saveCroppedImage"
+            auto-sizing
+            show-loading
+            prevent-white-space>
+          </croppa>
+          <v-card-actions class="img-actions">
+            <v-btn :color="primaryColor" @click="doneEditing" flat large>
+              <span class="mdi mdi-image"></span>
+              Done editing
+            </v-btn>
+            <v-tooltip bottom>
+              <template v-slot:activator="{ on }">
+                <v-btn :color="dangerColor" @click="removeImage" v-on="on" flat large>
+                  <span class="mdi mdi-delete"></span>
+                </v-btn>
+              </template>
+              <span>Delete image</span>
+            </v-tooltip>
+          </v-card-actions>
+        </v-card>
+      </v-flex>
+      <v-flex md8 class="flex-general">
+        <v-card class="card-general">
+          <meta-input
+            v-for="it in requiredData"
+            :meta="it"
+            :key="it.key"
+            @update="updateKey"
+            class="meta-input">
+          </meta-input>
+          <v-card-actions class="form-actions">
+            <v-btn @click="routeTo('catalog')" flat large>Cancel</v-btn>
+          </v-card-actions>
+        </v-card>
+      </v-flex>
+    </v-layout>
+    <v-snackbar v-model="snackbar" :timeout="timeout" :color="dangerColor" left>
+      There is no any file. Please choose an image.
+      <v-btn @click="snackbar=false" dark flat>Close</v-btn>
+    </v-snackbar>
+  </v-container>
+</template>
+
+<script>
+import Meta from 'components/common/Meta';
+import cloneDeep from 'lodash/cloneDeep';
+import set from 'lodash/set';
+import assetsApi from '../../../api/asset.js';
+import { mapActions, mapGetters } from 'vuex-module';
+
+const validate = { required: true, min: 2, max: 250 };
+
+export default {
+  name: 'user-settings',
+  data() {
+    return {
+      snackbar: false,
+      timeout: 2500,
+      dangerColor: '#d9534f',
+      primaryColor: '#337ab7',
+      croppa: {}
+    };
+  },
+  computed: {
+    ...mapGetters(['user']),
+    requiredData() {
+      return [{
+        key: 'role',
+        editing: false,
+        value: this.user.role,
+        type: 'INPUT',
+        label: 'Role',
+        validate
+      }, {
+        key: 'email',
+        editing: true,
+        value: this.user.email,
+        type: 'INPUT',
+        label: 'Email',
+        validate
+      }, {
+        key: 'firstName',
+        editing: true,
+        value: this.user.firstName,
+        type: 'INPUT',
+        label: 'First name',
+        validate
+      }, {
+        key: 'lastName',
+        editing: true,
+        value: this.user.lastName,
+        type: 'INPUT',
+        label: 'Last name',
+        validate
+      }];
+    }
+  },
+  methods: {
+    ...mapActions(['updateInfo']),
+    updateKey(key, value) {
+      const data = cloneDeep(this.user);
+      this.updateInfo(set(data, key, value));
+    },
+    saveCroppedImage(file) {
```

Comment:
Profile photo handling actions still in progress. I read croppa docs, still little confused on how to set image permanently (after refresh). I managed to send image data to server, then the multer does his job and returns the object with the key, url, and publicUrl. Furthermore, I still do not know how to extract an img element from croppa. Because of that I can not set url from the returned response on image src. Any suggestions?